etls/raw/atena/gc_customer_importer.py

```
@@ -82,17 +96,30 @@
     gc.write.mode("overwrite").parquet("gs://prd-lake-raw-atena/gc_customer/")
 
 
-# Imports the Table
-ind_df = spark.read.option("delimiter", "|") \
-    .csv("gs://prd-lake-transient-atena/atena/gc_cliente_indice/")
+# Source of the GC Customers products/
+gc_ind_source_path = \
+    "gs://prd-lake-transient-atena/atena/" \
+    "gc_cliente_indice/crm_gc_cliente_indice_{}.txt.gz" \
+    .format(lastday)
+
+# Loading products
+foundGCInd = True
+try:
+    ind_df = spark.read.option("delimiter", "|") \
+        .csv(gc_ind_source_path)
+except:
+    foundGCInd = False
+
+# If exists, processing products
+if foundGCInd:
```

why isn't this inside your try/except?

---

It always searches for a specific file ... If for some reason the file of the day did not arrive in the transient it treats the error by not finding it

---

What I mean is that the whole structure should be inside your try/except, instead of using a flag in the exception session and if this file is mandatory in your processing it should raise an exception and fail the pipeline.

```
    ind = spark.read.option("delimiter", "|") \
        .csv(gc_ind_source_path).select(
        col("_c0").alias("idorigem"),
        col("_c1").alias("origem"),
        col("_c2").alias("numdbm")) \
        .withColumn("datalog", lit(datetime.date.today()))
```